kornia/feature/harris.py

```diff
@@ -74,42 +85,70 @@ def forward(self, input: torch.Tensor) -> torch.Tensor:  # type: ignore
                              .format(input.shape))
         # compute the first order gradients with sobel operator
         # TODO: implement support for kernel different than three
-        gradients: torch.Tensor = spatial_gradient(input)
+        grad_mode = 'Sobel' if self.do_blur else 'NoBlur'
+        gradients: torch.Tensor = spatial_gradient(input, grad_mode)
         dx: torch.Tensor = gradients[:, :, 0]
         dy: torch.Tensor = gradients[:, :, 1]
 
         # compute the structure tensor M elements
         def g(x):
             return gaussian_blur2d(x, (3, 3), (1., 1.))
-
         dx2: torch.Tensor = g(dx * dx)
         dy2: torch.Tensor = g(dy * dy)
         dxy: torch.Tensor = g(dx * dy)
-
         det_m: torch.Tensor = dx2 * dy2 - dxy * dxy
         trace_m: torch.Tensor = dx2 + dy2
 
         # compute the response map
-        scores: torch.Tensor = det_m - self.k * trace_m ** 2
+        scores: torch.Tensor = det_m - self.k * (trace_m ** 2)
 
         # threshold
         # TODO: add as signature parameter ?
-        scores = torch.clamp(scores, min=1e-6)
+        scores = torch.clamp(scores, min=0)
 
         # apply non maxima suppresion
-        scores = non_maxima_suppression2d(scores, kernel_size=(3, 3))
+        if self.nms:
+            scores = non_maxima_suppression2d(scores, kernel_size=(3, 3))
 
         # normalize and return
-        scores_max: torch.Tensor = F.adaptive_max_pool2d(scores, output_size=1)
-        return scores / scores_max
+        if self.normalize:
+            scores_max: torch.Tensor = F.adaptive_max_pool2d(scores,
+                                                             output_size=1)
+            scores = scores / scores_max.clamp(1e-6)
```

Comment:
1) be explicit in what we clamp: `scores = scores / torch.clamp(scores_max, min=1e-6)`: I tend to use torch.clamp, torch.squeeze instead of using built-in tensor functions. Just to be consistent with the existing code.
2) should 1e-6 be an eps parameter ?

Reply:
I strongly disagree about 1) 
For consistency we then should use it verywhere. 
Imagine smth like. torch.sum(torch.pow(x,2), dim = 1) instead of simple x.pow(2).sum(dim=1)

Reply:
mmm, okay. Like it more in my way but it's fine